src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -0,0 +1,874 @@
+                    group + ", " + name + ") which is non-existent.");
+        }
+        return QJSValue();
+    }
+
+    if (!callback.isCallable()) {
+        m_pScriptEngineLegacy->throwJSError("Tried to connect (" + group + ", " + name +
+                ")" +
+                " to an invalid callback. Make sure that your code contains no "
+                "syntax errors.");
+        return QJSValue();
+    }
+
+    ScriptConnection connection;
+    connection.key = ConfigKey(group, name);
+    connection.engineJSProxy = this;
+    connection.controllerEngine = m_pScriptEngineLegacy;
+    connection.callback = callback;
+    connection.id = QUuid::createUuid();
+
+    if (coScript->addScriptConnection(connection)) {
+        return pJsEngine->newQObject(
+                new ScriptConnectionJSProxy(connection));
+    }
+
+    return QJSValue();
+}
+
+bool ControllerScriptInterfaceLegacy::removeScriptConnection(
+        const ScriptConnection connection) {
+    ControlObjectScript* coScript =
+            getControlObjectScript(connection.key.group, connection.key.item);
+
+    if (m_pScriptEngineLegacy->jsEngine() == nullptr || coScript == nullptr) {
+        return false;
+    }
+
+    return coScript->removeScriptConnection(connection);
+}
+
+void ControllerScriptInterfaceLegacy::triggerScriptConnection(
+        const ScriptConnection connection) {
+    VERIFY_OR_DEBUG_ASSERT(m_pScriptEngineLegacy->jsEngine()) {
+        return;
+    }
+
+    ControlObjectScript* coScript =
+            getControlObjectScript(connection.key.group, connection.key.item);
+    if (coScript == nullptr) {
+        return;
+    }
+
+    connection.executeCallback(coScript->get());
+}
+
+// This function is a legacy version of makeConnection with several alternate
+// ways of invoking it. The callback function can be passed either as a string of
+// JavaScript code that evaluates to a function or an actual JavaScript function.
+// If "true" is passed as a 4th parameter, all connections to the ControlObject
+// are removed. If a ScriptConnectionInvokableWrapper is passed instead of a callback,
+// it is disconnected.
+// WARNING: These behaviors are quirky and confusing, so if you change this function,
+// be sure to run the ControllerScriptInterfaceTest suite to make sure you do not break old scripts.
+QJSValue ControllerScriptInterfaceLegacy::connectControl(
+        QString group, QString name, QJSValue passedCallback, bool disconnect) {
+    // The passedCallback may or may not actually be a function, so when
+    // the actual callback function is found, store it in this variable.
+    QJSValue actualCallbackFunction;
+
+    if (passedCallback.isCallable()) {
+        if (!disconnect) {
+            // skip all the checks below and just make the connection
+            return makeConnection(group, name, passedCallback);
+        }
+        actualCallbackFunction = passedCallback;
+    }
+
+    auto pScriptEngine = m_pScriptEngineLegacy->jsEngine();
```

Comment:
```suggestion
    auto pJsEngine = m_pScriptEngineLegacy->jsEngine();
```

Else it is too similar to m_pScriptEngineLegacy

Reply:
good catch. Fixed 4227e22